assets/js/modules/analytics-4/datastore/settings.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * `modules/analytics-4` data store: settings.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import { createStrictSelect } from '../../../googlesitekit/data/utils';
+import { isValidPropertySelection, isValidWebDataStreamID } from '../utils/validation';
+import { invalidateCacheGroup } from '../../../components/data/invalidate-cache-group';
+import { INVARIANT_DOING_SUBMIT_CHANGES, INVARIANT_SETTINGS_NOT_CHANGED } from '../../../googlesitekit/data/create-settings-store';
+import { MODULES_ANALYTICS } from '../../analytics/datastore/constants';
+import { STORE_NAME, PROPERTY_CREATE } from './constants';
+import { TYPE_MODULES } from '../../../components/data';
+
+// Invariant error messages.
+export const INVARIANT_INVALID_PROPERTY_SELECTION = 'a valid propertyID is required to submit changes';
+export const INVARIANT_INVALID_WEBDATASTREAM_ID = 'a valid webDataStreamID is required to submit changes';
+
+export async function submitChanges( { select, dispatch } ) {
+	let propertyID = select( STORE_NAME ).getPropertyID();
+	if ( propertyID === PROPERTY_CREATE ) {
+		const accountID = select( MODULES_ANALYTICS ).getAccountID();
+		const { response: property, error } = await dispatch( STORE_NAME ).createProperty( accountID );
+		if ( error ) {
+			return { error };
+		}
+
+		propertyID = property._id;
+		await dispatch( STORE_NAME ).setPropertyID( propertyID );
+		await dispatch( STORE_NAME ).setWebDataStreamID( '' );
```

Comment:
Why do we need to set the webstream data ID as an empty string here? We select it and then check to see if it's valid, below, so presumably a `null`/`undefined` value or similar won't be valid, right?

Or does it need to be a string, but an invalid string, in order for the `createWebDataStream` path below to work? If so, it's worth adding a comment here to clarify that. At any rate it isn't obvious to me why this line is here.

Reply:
We need to set an invalid web data stream ID to make sure a new data stream will be created. I have added a comment to explain it.